pkg/kodo/cmd/build.go

```diff
@@ -0,0 +1,116 @@
+package cmd
+
+import (
+	"context"
+
+	"k8s.io/client-go/rest"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	buildv1api "github.com/openshift/api/build/v1"
+	imagev1api "github.com/openshift/api/image/v1"
+
+	buildv1clientapi "github.com/openshift/client-go/build/clientset/versioned/typed/build/v1"
+	imagev1clientapi "github.com/openshift/client-go/image/clientset/versioned/typed/image/v1"
+)
+
+func createTypeMeta(kind string, APIVersion string) metav1.TypeMeta {
+	return metav1.TypeMeta{
+		Kind:       kind,
+		APIVersion: APIVersion,
+	}
+}
+
+func createObjectType(name string, namespace string) metav1.ObjectMeta {
+	if namespace != "" {
+		return metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		}
+	} else {
```

Comment:
Would not setting the namespace be permissible in any scenario?

Reply:
BuildConfig doesn't have a Namespace specified in the description of issue, hence added the condition.

Reply:
BuildConfigs are inherently namespace-scoped resources. I didn't specify in the issue because my expectation was that it would be specified on the fly :)

Reply:
Noted, will make the change and push again

Reply:
Changes have been made